### PR TITLE
Add support for explicitly setting modified times

### PR DESCRIPTION
**Pull request by mboynes**
This also includes a minor refactor of the unit tests to DRY them up.

### CODE REVIEW

File: inc/class-rest-api.php [hunk 1]
```diff
@@ -60,6 +60,120 @@ public function register_routes() {
...
+	/**
+	 * Add filters for modifying core data or functionality, but only during SST
+	 * REST requests.
+	 */
+	public function add_sst_request_filters() {
+		foreach ( get_post_types() as $post_type ) {
+			add_filter( "rest_pre_insert_{$post_type}", [ $this, 'set_modified' ], 10, 2 );
+		}
+		add_filter( 'wp_insert_post_data', [ $this, 'prevent_updates_from_overwriting_modified_date' ], 5, 2 );
+		add_filter( 'wp_insert_post_data', [ $this, 'set_modified_for_real' ], 10, 2 );
```
Comment (scottnelle) [inline inc/class-rest-api.php:72]: 😆

File: tests/test-rest-api.php [hunk 2]
```diff
@@ -539,4 +480,62 @@ public function test_post_with_featured_image() {
...
+
+	public function test_set_modified_date() {
+		$modified = '2001-02-03 04:05:06';
+
+		$data = $this->create_post_request_with_defaults( compact( 'modified' ) );
+
+		$this->assertFalse( empty( $data['posts'][0]['post_id'] ) );
+		$post_id = $data['posts'][0]['post_id'];
+
+		// Confirm that the image was stored as the post thumbnail.
```
Comment (scottnelle) [inline tests/test-rest-api.php]: Looks like this comment was copied from another test and isn't applicable here.
```suggestion
		// Confirm that the post modified time was set correctly.
```

File: tests/test-rest-api.php [hunk 3]
```diff
@@ -539,4 +480,62 @@ public function test_post_with_featured_image() {
...
+		$this->assertSame(
+			$modified,
+			get_post_modified_time( 'Y-m-d H:i:s', false, $post_id )
+		);
+
+		// Ensure that subsequently updates don't modify the post modified time.
```
Comment (scottnelle) [inline tests/test-rest-api.php]:
```suggestion
		// Ensure that subsequent updates don't modify the post modified time.
```

**Review by scottnelle — APPROVED** [review]
A couple of suggestions relating to comments. 👻

File: inc/class-rest-api.php (hunk 1 above)
Comment (mboynes) [reply inc/class-rest-api.php:72]: I was really hoping @jakewrfoster would grab the code review for this joke

**Pull request merged**